JsonSubTypes.Tests/AbstractBaseClassDiscriminatorTests.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Newtonsoft.Json;
+using NUnit.Framework;
+
+namespace JsonSubTypes.Tests
+{
+    [TestFixture]
+    public class AbstractBaseClassDiscriminatorTests
+    {
+        [JsonConverter(typeof(JsonSubtypes), nameof(SubTypeClass.Discriminator))]
+        public abstract class MainClass
+        {
+        }
+
+        public class SubTypeClass : MainClass
+        {
+            public string Discriminator => "SubTypeClass";
+        }
+
+        [Test]
+        [Timeout(2000)]
+        public void DeserializingWithAbstractBaseClassDiscriminatorThrows()
+        {
```

Comment:
Thanks for providing non regression tests !

What about asserting the message too ?

```cs
            var exception = Assert.Throws<JsonSerializationException>(() => JsonConvert.DeserializeObject<MainClass>("{\"Discriminator\":\"MainClass\"}"));
            Assert.AreEqual("Could not create an instance of type JsonSubTypes.Tests.AbstractBaseClassDiscriminatorTests+MainClass. Type is an interface or abstract class and cannot be instantiated. Path 'Discriminator', line 1, position 17.", exception.Message);
```

Reply:
Let's do that in the same commit where we remove the code that throws a custom "circular reference" exception.